src/Libraries/Microsoft.Extensions.AI.OpenAI/OpenAISerializationHelpers.cs

```diff
@@ -0,0 +1,134 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Buffers;
+using System.ClientModel.Primitives;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.ServerSentEvents;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Shared.Diagnostics;
+using OpenAI.Chat;
+
+#pragma warning disable S1135 // Track uses of "TODO" tags
+
+namespace Microsoft.Extensions.AI;
+
+/// <summary>
+/// Defines a set of helpers used to serialize Microsoft.Extensions.AI content using the OpenAI wire format.
+/// </summary>
+public static class OpenAISerializationHelpers
+{
+    /// <summary>
+    /// Deserializes a stream using the OpenAI wire format into a pair of <see cref="ChatMessage"/> and <see cref="ChatOptions"/> values.
+    /// </summary>
+    /// <param name="stream">The stream containing a message using the OpenAI wire format.</param>
+    /// <param name="options">The <see cref="JsonSerializerOptions"/> governing deserialization of function call content.</param>
+    /// <param name="cancellationToken">A token used to cancel the operation.</param>
+    /// <returns>The deserialized list of chat messages and chat options.</returns>
+    public static async Task<(IList<ChatMessage> Messages, ChatOptions? Options)> DeserializeFromOpenAIAsync(
```

Comment:
Is this about deserializing a request, deserializing a non-streaming response, or deserializing a streaming response?

Reply:
It's for deserializing requests. API names are largely placeholders, will refine.